nemo/collections/asr/parts/utils/nmse_clustering.py

```diff
@@ -315,38 +390,43 @@ def getPvalueList(self):
 
         return p_value_list
 
-    def estimateNumofSpeakers(self, X_conn):
-        """Estimate the number of speakers using eigen decompose on Laplacian Matrix.
+    def getLamdaGaplist(self, lambdas):
+        lambdas = np.real(lambdas)
+        return list(lambdas[1:] - lambdas[:-1])
+
+    def estimateNumofSpeakers(self, affinity_mat):
+        """
+        Estimates the number of speakers using eigen decompose on Laplacian Matrix.
         """
-        Laplacian = getLaplacian(X_conn)
+        Laplacian = getLaplacian(affinity_mat)
         if self.cuda:
-            lambdas = eigDecomposeGPU(Laplacian)
+            lambdas, _ = eigDecomposeGPU(Laplacian)
```

Comment:
You can actually have only one function `eigDecompose` and send Laplacian matrix a cuda tensor or cpu tensor if cuda device available.